src/olympia/files/models.py

```diff
@@ -183,9 +185,11 @@ def from_upload(cls, upload, version, platform, parsed_data=None):
             # Add content_scripts host matches too.
             for script in parsed_data.get('content_scripts', []):
```

Comment:
I see that we extend `permissions` with other manifest properties here. Is there any optional permission elsewhere in the `manifest` file? cc @Rob--W

Reply:
Only `optional_permissions` contains permissions.

Does the code here behave correctly when passed an invalid value for the key or array?

And since it's related (to mandatory permissions), does the code also account for the devtools_page manifest key (which triggers a permission warning)?

Reply:
> Does the code here behave correctly when passed an invalid value for the key or array?

Would you have an example in mind?

> And since it's related (to mandatory permissions), does the code also account for the devtools_page manifest key (which triggers a permission warning)?

yeah, we handle that.

Reply:
```
// Invalid, rejected by Firefox:
"optional_permissions": 123,
"optional_permissions": { },

// Accepted by Firefox, but unsupported values are ignored:
"optional_permissions": null,
"optional_permissions": [1234],
"optional_permissions": [{"__proto__": {}}],
```

Reply:
Oh I see now, thanks! We'll likely add some more test cases to make sure that we can handle those values one way or another.

Reply:
Code has been added to ignore any invalid optional permissions: https://github.com/mozilla/addons-server/pull/15047/files#diff-ecb8df04bee7aac11123f3729e23cfafR366-R373, and a test was added as well. It covers all the cases above except for `null`, so I'll add that to the test.